icu4c/source/i18n/msgfmt.cpp

```diff
@@ -810,26 +810,31 @@ MessageFormat::getFormats(int32_t& cnt) const
     // method on this object.  We construct and resize an array
     // on demand that contains aliases to the subformats[i].format
     // pointers.
+
+    // Get total required capacity first (it's refreshed on each call).
+    int32_t total_capacity = 0;
+    for (int32_t partIndex = 0; (partIndex = nextTopLevelArgStart(partIndex)) >= 0; ++total_capacity);
```

Comment:
Optional (but would help readability):
```
for (...);
```
-->
```
for (...) {}
```
Much easier to spot an empty loop body this way.

Reply:
Done